docs/intro/overview.rst

```diff
@@ -13,172 +13,83 @@ precisely, `web scraping`_), it can also be used to extract data using APIs
-
-   ...
-
-.. highlight:: none
-
-An XPath expression to select the description could be::
 
-    //div[@id='description']
+    class StackOverflowSpider(scrapy.Spider):
+        name = 'stackoverflow'
+        start_urls = ['http://stackoverflow.com/questions?sort=votes']
 
-.. highlight:: html
+        def parse(self, response):
+            for href in response.css('.question-summary h3 a::attr(href)'):
+                full_url = response.urljoin(href.extract())
+                yield scrapy.Request(full_url, callback=self.parse_question)
 
-Finally, the file size is contained in the second ``<p>`` tag inside the ``<div>``
-tag with ``id=specifications``::
+        def parse_question(self, response):
+            title = response.css('h1 a::text').extract_first()
+            votes = response.css('.question .vote-count-post::text').extract_first()
+            tags = response.css('.question .post-tag::text').extract()
+            body = response.css('.question .post-text').extract_first()
+            yield {
+                'title': title,
+                'votes': votes,
+                'body': body,
+                'tags': tags,
+                'link': response.url,
+            }
 
-   <div id="specifications">
 
-   <p>
-   <strong>Category:</strong>
-   <a href="/cat/4">Movies</a> &gt; <a href="/sub/35">Documentary</a>
-   </p>
+Put this in a file, name it to something like ``stackoverflow_spider.py``
+and run the spider using the :command:`runspider` command::
 
-   <p>
-   <strong>Total size:</strong>
-   150.62&nbsp;megabyte</p>
+    scrapy runspider stackoverflow_spider.py -o top-stackoverflow-questions.json
 
 
-.. highlight:: none
+When this finishes you will have in the ``top-stackoverflow-questions.json`` file
+a list of the most upvoted questions in StackOverflow in JSON format, containing the
+title, link, number of upvotes, a list of the tags and the question content in HTML.
 
-An XPath expression to select the file size could be::
 
-   //div[@id='specifications']/p[2]/text()[2]
+What just happened?
+-------------------
 
-.. highlight:: python
+When you ran the command ``scrapy runspider somefile.py``, Scrapy looked
+for a Spider definition inside it and ran it through its crawler engine.
 
-For more information about XPath see the `XPath reference`_.
+The crawl started by making requests to the URLs defined in the ``start_urls``
+attribute (in this case, only the URL for StackOverflow top questions page),
+and then called the default callback method ``parse`` passing the response
+object as an argument.
 
-Finally, here's the spider code::
+In the ``parse`` callback, we scrape the links to the questions and
+yield a few more requests to be processed, registering for them
+the method ``parse_question`` as the callback to be called when the
+requests are complete.
```

Comment:
I think here we should explain that these requests are scheduled and processed asynchronously.